app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+
+  def self.update_posts_counter(user_id)
+    user = User.find(user_id)
+    user.posts.update_all(comments_counter: user.posts.joins(:comments).distinct.count)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```

Comment:
- [ ]  Great job for creation the the association between user and the posts, but you should also add the association between the posts and the likes.  because it is possible that a post might have  likes . One more other things I recommend you use` author` instead of `user`, and user  `User `as a `class_name`

- [ Optional]  Your update_post_counter method could be improved, to reduce it's complexity and the user of userId, by just  considering that You have the User in your object scope(ActiveRecord ) and use` increment!`  ActiveRecord method that increments the attribute by 1 and saves the record.